### PR TITLE
Allow filtering of connectors on connector selection page

### PR DESCRIPTION
**Pull request by kuosandys**
Towards https://github.com/giantswarm/roadmap/issues/1592.

This PR changes `dex-app` to use a custom container image for dex (built from the Dockerfile in this repo), so that we could use our own web templates for dex. ([Relevant docs page from dex](https://dexidp.io/docs/templates/))

The files in the `web` directory are a copy of dex's original template files, with the exception of the `login.html` template, which includes some added JS to allow us to filter connectors on the connector selection page (`/auth`).

Connectors can be filtered by specifying the `connector_filter` query parameter:
- if a filter is specified, only connectors with ID's that start with the specified filter will be displayed
- if a filter is not specified, all connectors will be displayed
- if a user has disabled JS for their browser, all connectors will be displayed

## Checklist

- [ ] Update CHANGELOG.md

### CODE REVIEW

File: web/templates/login.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{{ template "header.html" . }}
```
Comment (kuosandys) [inline web/templates/login.html:1]: For reference, [this](https://github.com/dexidp/dex/blob/master/web/templates/login.html) is the `login.html` template from dex's repo.

The JS executing the connector filtering logic is contained within the `<script />` tags. For ease of review, I haven't minimized it.

**Review by marians — APPROVED** [review]
Looks great from what I understand.

**Review by gusevda — APPROVED** [review]
Looks great

**Review by vvondruska — APPROVED** [review]
Looks good to me. Tested the flow between kubectl-gs and Dex with multiple connectors and it worked fine for me.

**Pull request merged**